File: Contest/Biweekly/Biweekly Contest 170/3751. Total Waviness of Numbers in Range I/Solution.java
import java.util.*;

class Solution {
    public int totalWaviness(int num1, int num2) {
        // N : no of number btw num1 and num2, M : no of digits
        // Time Complexity: O(N * M)
        // Space Complexity: O(M)
        int totalWav = 0;
        for (int num = num1; num <= num2; num++) {
            String numStr = Integer.toString(num);
            for (int i = 1; i < numStr.length() - 1; i++) {
                if ((Character.getNumericValue(numStr.charAt(i)) < Character.getNumericValue(numStr.charAt(i - 1)) && 
                Character.getNumericValue(numStr.charAt(i)) < Character.getNumericValue(numStr.charAt(i + 1))) || 
                (Character.getNumericValue(numStr.charAt(i)) > Character.getNumericValue(numStr.charAt(i - 1)) && 
                Character.getNumericValue(numStr.charAt(i)) > Character.getNumericValue(numStr.charAt(i + 1)))) {
                    totalWav++;
                }
            }
        }
        return totalWav;
    }
}

File: Contest/Biweekly/Biweekly Contest 170/3751. Total Waviness of Numbers in Range I/Solution.py
class Solution:
    def totalWaviness(self, num1: int, num2: int) -> int:
        # N : no of number btw num1 and num2, M : no of digits
        # Time Complexity: O(N * M)
        # Space Complexity: O(M)
        totalWav = 0
        for num in range(num1, num2 + 1):
            s = str(num)
            for i in range(1, len(s) - 1):
                if s[i - 1] < s[i] and s[i] > s[i + 1]:
                    totalWav += 1
                elif s[i - 1] > s[i] and s[i] < s[i + 1]:
                    totalWav += 1
        return totalWav